testsuite/model/src/test/java/org/keycloak/testsuite/model/globalLock/GlobalLocksTest.java

```diff
@@ -0,0 +1,174 @@
+
+            try {
+                lockProvider.withLock(LOCK_NAME, Duration.ofSeconds(60), innerSession -> {
+                    LOG.infof("Iteration %d entered locked block", index);
+
+                    // Locked block
+                    int c = counter.getAndIncrement();
+
+                    try {
+                        Thread.sleep(rand.nextInt(100));
+                    } catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                        throw new RuntimeException(e);
+                    }
+
+                    resultingList.add(c);
+                    return null;
+                });
+            } catch (LockAcquiringTimeoutException e) {
+                throw new RuntimeException(e);
+            }
+        }));
+
+        assertThat(resultingList, hasSize(numIterations));
+        assertThat(resultingList, equalTo(IntStream.range(0, 50).boxed().collect(Collectors.toList())));
+    }
+
+    @Test
+    public void lockTimeoutExceptionTest() {
+        final String LOCK_NAME = "lockTimeoutExceptionTestLock";
+        AtomicInteger counter = new AtomicInteger();
+        CountDownLatch waitForTheOtherThreadToFail = new CountDownLatch(1);
+
+        IntStream.range(0, 2).parallel().forEach(index -> inComittedTransaction(s -> {
+            GlobalLockProvider lockProvider = s.getProvider(GlobalLockProvider.class);
+
+            try {
+                lockProvider.withLock(LOCK_NAME, Duration.ofSeconds(2), innerSession -> {
+                    int c = counter.incrementAndGet();
+                    if (c == 1) {
+                        try {
+                            waitForTheOtherThreadToFail.await();
+                        } catch (InterruptedException e) {
+                            throw new RuntimeException(e);
+                        }
+                    } else {
+                        LOG.infof("Lock acquired by thread %s with counter: %d", Thread.currentThread().getName(), c);
+                        throw new RuntimeException("Lock acquired by more than one thread.");
+                    }
+                    return  null;
+                });
+            } catch (LockAcquiringTimeoutException e) {
+                int c = counter.incrementAndGet();
+                LOG.infof("Exception when acquiring lock by thread %s with counter: %d", Thread.currentThread().getName(), c);
+                if (c != 2) {
+                    throw new RuntimeException("Acquiring lock failed by different thread than second.");
+                }
+
+                assertThat(e.getMessage(), containsString("Lock [" + LOCK_NAME + "] already acquired by keycloak instance"));
+                waitForTheOtherThreadToFail.countDown();
+            }
+        }));
+    }
+
+    @RequireProvider(value = GlobalLockProvider.class, only = HotRodGlobalLockProviderFactory.PROVIDER_ID)
+    @Test
+    public void testReleaseAllLocksForHotRodLocks() {
```

Comment:
Can this be run without the limitation of the `@RequireProvider` above?

e.g. two threads - one acquires "some locks" `LOCK_1` .. `LOCK_9`, then waits for the other to perform the test, and then releases the lock? It would resemble the real scenario better.